dev/core/src/com/google/gwt/core/ext/linker/impl/installScriptEarlyDownload.js

```diff
@@ -19,23 +19,15 @@ function installScript(filename) {
     var doc = getInstallLocationDoc();
     var docbody = doc.body;
     var script;
-    // for sourcemaps, we inject textNodes into the script element on Chrome
-    if (navigator.userAgent.indexOf("Chrome") > -1 && window.JSON) {
-      var scriptFrag = doc.createDocumentFragment()
-      // surround code with eval until crbug #90707 
-      scriptFrag.appendChild(doc.createTextNode("eval(\""));
+    // for sourcemaps, we inject the code as a single string for Chrome
```

Comment:
Shouldn't we do this for all browsers as sourcemaps are now supported everywhere?

Reply:
Can I do that in a later PR? I'd want to perf test in safari and ff since it's a change in how many chunks the scripts are evaluated in

Reply:
It seems likely that with the general demise of IE that we don't even need these chunks any more, and can default to no limit, so that all browsers get this as a single string (without even a need for the join).

Reply:
Oh so that's IE related? Neato. I found [this review](https://github.com/gwtproject/gwt/commit/4814ffae4a81d396a2650d46f0b3c120bb69b926) but I've been having trouble getting more context around the change, is there something I can read up on?
Can that also be in another PR that I'll do? I'd want to do a bunch more testing around that change.

Reply:
The commit you link to shows that this was an IFrameLinker thing that had been ported to xsiframe. Looking at the IFrameLink history, we can find https://github.com/gwtproject/gwt/commit/7bcb57ae4fd79ffd0eba43dda335fda8c8c0458f (nearly 14 years old):

> This patch modifies the IFrame linker to use multiple script tags instead of one giant one.  This improves parsing time on Firefox, and it should improve overall loading latency on any browser whenever the network latency is significant.

I'm not sure how this applies to xsiframe though, as the chunks aren't streamed over the network (contrary to HTML in the iframe linker), unless it was part of an experiment at Google at the time to somehow use a .cache.html with an xsiframe subclass?